streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStoreTest.java

```diff
@@ -178,44 +178,36 @@ public void shouldPutAndFetch() {
 
         try (final KeyValueIterator<Bytes, byte[]> values = bytesStore.fetch(
             Bytes.wrap(keyA.getBytes()), 0, windows[2].start())) {
-
-            final List<KeyValue<Windowed<String>, Long>> expected = asList(
-                KeyValue.pair(new Windowed<>(keyA, windows[0]), 10L),
-                KeyValue.pair(new Windowed<>(keyA, windows[1]), 50L)
-            );
+            // For all tests, actualFrom is computed using observedStreamTime - retention + 1.
+            // so actualFrom = 60000(observedStreamTime) - 1000(retention) + 1 = 59001
+            // all records expired as actual from is 59001 and to is 1000
+            final List<KeyValue<Windowed<String>, Long>> expected = Collections.emptyList();
```

Comment:
Should we instead update the query time range in this test to get some results instead of empty list? Or do we test other ranges later and just expecting no result is the best thing to do here?

Reply:
The reason I left it like this was that this same test has different fetches across different time ranges. Some of them don't return anything due to expiry and some of them do. So, I felt this test captures all those scenarios properly. Probably at this point, leaving it this way might be ok unless you have reservations on that idea.